zulipterminal/model.py

```diff
@@ -564,6 +565,60 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def get_latest_message_in_topic(
+        self, stream_id: int, topic_name: str
+    ) -> Dict[str, Any]:
+        request: Dict[str, Any] = {
+            "anchor": "newest",
+            "num_before": 1,
+            "num_after": 0,
+            "narrow": [
+                {
+                    "operator": "stream",
+                    "operand": self.stream_dict[stream_id]["name"],
+                },
+                {"operator": "topic", "operand": topic_name},
+            ],
+        }
+        response = self.client.get_messages(request)
+        display_error_if_present(response, self.controller)
+        return response
+
+    def can_user_edit_topic(self) -> bool:
+        if not self.initial_data.get("realm_allow_message_editing", None):
+            self.controller.report_error(" Editing messages is disabled")
+            return False
+        elif self.initial_data.get("realm_allow_community_topic_editing", None):
+            self.controller.report_error(" Editing topic is disabled")
+            return False
+        else:
+            return True
+
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> None:
+        if self.can_user_edit_topic():
+            response = self.get_latest_message_in_topic(stream_id, topic_name)
+            if response["messages"] != []:
+                time_since_msg_sent = time.time() - response["messages"][0]["timestamp"]
+                edit_time_limit = self.initial_data[
+                    "realm_community_topic_editing_limit_seconds"
```

Comment:
Note that as per the API changelog, this parameter was new in feature level 11, so we need a backup value for this.

In an ideal world we could put all of the error conditions in one function, but let's keep them here for now.